src/ring_theory/roots_of_unity.lean

```diff
@@ -650,35 +675,40 @@ end
 
 /-- `nth_roots n` as a `finset` is equal to the union of `primitive_roots i R` for `i ∣ n`
 if there is a primitive root of unity in `R`. -/
-lemma nth_roots_one_eq_bind_primitive_roots {ζ : R} {n : ℕ+} (h : is_primitive_root ζ n) :
-  nth_roots_finset n R = (nat.divisors ↑n).bind (λ i, (primitive_roots i R)) :=
+lemma nth_roots_one_eq_bind_primitive_roots {ζ : R} {n : ℕ} (hpos : 0 < n)
```

Comment:
It seems that the proof is becoming 5 lines longer? So an alternative approach is to state your version as `nth_roots_one_eq_bind_primitive_roots'` (with a `'`) and prove it using the original version. Seems like the total amount of new lines is the same, and later users will have more choice in which version they want to use.